### PR TITLE
Reclaim scala package

### PR DESCRIPTION
**Pull request by softprops**
this relates to https://github.com/bazelbuild/rules_scala/issues/526 ( specifically [this comment](https://github.com/bazelbuild/rules_scala/issues/526#issuecomment-398473801) ). The base package of the entire test hardness is `scala` which just happens to be the base package of scala std lib. This prevents the ability do refer to things like `scala.Option` within mixed java sources.

### CODE REVIEW

**Comment by softprops** [comment]
note this looks like a lot ( and it is ) but its exclusively renames of the test harness packaging from scala to scalarules.

File: test/BUILD [hunk 1]
```diff
@@ -236,18 +236,18 @@ scala_library(
...
     deps = ["MixJavaScalaLib"],
 )
 
 #Mix java scala (srcjar), much like above only the java is provided via srcjar
 scala_library(
     name = "MixJavaScalaSrcjarLib",
     srcs = glob([
-        "src/main/scala/scala/test/mix_java_scala/*.scala",
+        "src/main/scala/scalarules/test/mix_java_scala/*.scala",
         # srcjar created with `jar -cfM Baz.srcjar Baz.java`
```
Comment (softprops) [inline test/BUILD]: these comments were well with the extra few seconds it took to add them. I dealt with a number of failing tests related to jars that I had to recreate using the commentary in these rules.

**Comment by softprops** [comment]
Looks like a failing [scala_specs2_exception_in_initializer_without_filter test](https://travis-ci.org/bazelbuild/rules_scala/jobs/394390812#L874). Ill look into that.

**Comment by softprops** [comment]
@johnynek or others it looks like the builds is passing expect for one item on the build matrix which I can't tell if its related to my changes [here](https://travis-ci.org/bazelbuild/rules_scala/jobs/394576173#L237)

**Comment by johnynek** [comment]
thanks @softprops it looks like a transient download issue. restarting.

**Comment by johnynek** [comment]
thanks @softprops !

**Comment by ittaiz** [comment]
thanks!

**Comment by softprops** [comment]
Np. I think I can followup on the mixed compilation thing with a failing test case. I'll likely need to get up to speed on what's changed since I last understood the code base. In any case this exercise gave me a greater appreciation for the amount of testing in these rules. One thing may suggest is tips for contributors on getting faster turn around cycles. Rename / retest cycles for this particular pr were massive!

**Pull request merged**